src/options.js

```diff
@@ -70,6 +74,104 @@ function loadOptions() {
 $(loadOptions);
 
 /**
+ * Import a list of trackers supplied by the user
+ * NOTE: list must be in JSON format to be parseable
+ */
+function importTrackerList() {
+  var file = $('#inputFile')[0].files[0];
+
+  if (file) {
+    var reader = new FileReader();
+    reader.readAsText(file);
+    reader.onload = function(e) {
+      parseTrackerList(e.target.result);
+    };
+  } else {
+    confirm("Please select a file to import.");
+  }
+
+  document.getElementById("inputFile").value = '';
+}
+
+/**
+ * Parse the tracker lists uploaded by the user, adding to
+ * action_map and snitch_map anything that isn't currently present.
+ *
+ * @param trackerLists - data from JSON file that user provided
+ */
+function parseTrackerList(trackerLists) {
+  var lists;
+
+  try {
+    lists = JSON.parse(trackerLists);
+  } catch (e) {
+    confirm("File must be of type .json.");
+  }
+
+  for (var i in lists) {
+    var elem = lists[i];
+    if (elem.action_map) {
+      updateMap(elem.action_map, "action_map");
+    } else if (elem.snitch_map) {
+      updateMap(elem.snitch_map, "snitch_map");
+    }
+  }
+
+  // Update list to reflect new status of map
+  refreshFilterPage();
+  confirm("Tracker list updated successfully!");
+}
+
+/**
+ * Update the specified storage map with the elements
+ * from the given list. Supported maps are currently `action_map`
+ * and `snitch_map`
+ *
+ * @param list - the list of data to be added
+ * @param map - the map in which the data is to be inserted
+ */
+function updateMap(list, map) {
+  var storageMap;
+  if (map === "action_map") {
+    storageMap = badger.storage.getBadgerStorageObject("action_map");
+
+  } else if (map === "snitch_map") {
+    storageMap = badger.storage.getBadgerStorageObject("snitch_map");
```

Comment:
create mergeObject API on storage.js and then call badger.storage.getBadgerStorageObject("foo").mergeObject(list)